### ISSUE TITLE
scope of the project & more

### ISSUE THREAD

**Issue by ltalirz** [iheartla/iheartla#122]
Hi, 

I just came across this cool project as a reviewer of your NumFOCUS affiliated project application and wanted to share some thoughts that came to mind as someone completely new to iheartla.

 1.  The README expresses quite succinctly what iheartla aims to be. At the same time, it may be useful also state explicitly what you consider *out* of its scope, e.g. other fields of mathematics / writing complex programs / ...
     It seems like the requirement to produce "working code" could impose a significant limitation on the "mathematical sentences" one can express in iheartla, and it would be interesting to understand the reasoning behind this decision in terms of the use cases targeted by the language.
 1. Example use cases: it might be useful to provide a set of practical use cases that are already possible today (e.g. sharing equations via twitter, note taking, quickly prototyping the C++ code for an equation, ...), as well as others you see coming in the future. The current iheartla gallery seems to consist mainly of equations translated from scientific papers to iheartla, the practical use of which may not be immediately obvious. I guess in principle one could use this to jumpstart a program that uses the equation, but in order to take advantage of iheartla in scientific writing (which I understand to be the main implied use case here), one will first need support for iheartla in the authoring tools for manuscript/web pages/.... 
 1. I see that part of the codebase already uses mathjax. Is it a goal for iheartla to become supported as a first-level math language in mathjax? (this would make a lot of sense to me)
 1. While the wiki claims that entering Unicode characters "isn't that difficult", finding efficient ways of writing iheartla is obviously going to be very important for its adoption. It could be helpful to
    - showcase how *you* (the power users) efficiently enter Unicode
    - support existing standards, such as symbols from `amsmath` (which you already do to some degree) or [ASCIImath](http://asciimath.org/), so that users don't have to relearn shortcuts

These comments are intended merely as food for thought, and are separate from the evaluation of your application.

**Comment by yig** [comment]
These are great suggestions. Thank you! We are merging in code from the latest "paper" (conditionally accepted; the code is on the `local_func` branch). We will want to link to that paper, too. Of course, most people won't read a paper. We'll have to explain everything in the README. To answer the specific questions:

1. We have some some ideas about expanding I❤️LA to cover "non-executable" math. This would avoid needing to type LaTeX (the multi-language) problem and could open up possibilities for some kind of verification of derivations or symbol use in other contexts. It's not high on our to-do list, though. Covering all the mathematical notation (in all branches of math) that LaTeX covers is a non-goal. There's just too much and it strays too far from the executable goal of I❤️LA.
2. This is a good idea.
3. We hadn't considered this. Would the idea be that MathJax could nicely typeset I❤️LA, or also compile it into executable code? Compiling it into executable code would mean the entire compiler would become part of MathJax.
4. Personally, on my mac, I set up text replacements (System Preferences > Keyboard > Text) that replace latex commands (e.g., `^2` and `\theta`) with Unicode (`²` and `θ`). The wiki points users to a `plist` file they can drag and drop. Others do other things. Is this information buried, or do you suggest something like a video tutorial?

**Comment by ltalirz** [comment]
Thanks for the thoughtful replies.

Re 3.: I was only thinking of typesetting; essentially just like your gallery but the opposite direction (from iheartla to latex) and in the browser
